pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java

```diff
@@ -225,17 +225,17 @@ public void createPartitionedTopic(
             @PathParam("namespace") String namespace,
             @ApiParam(value = "Specify topic name", required = true)
             @PathParam("topic") @Encoded String encodedTopic,
-            @ApiParam(value = "The number of partitions for the topic",
-                    required = true, type = "int", defaultValue = "0")
-                    int numPartitions,
+            @ApiParam(value = "The metadata for the topic",
```

Comment:
Looks like a breaking change?

Reply:
For there exists one param in this method before, we can't add a new param to this method because the body value only supports one param.

Reply:
@codelipenghui It looks like not. I ran the following command in my local env:

```bash
curl -L http://localhost:8080/admin/v2/persistent/public/default/xxx/partitions -X PUT -H 'Content-Type: application/json' -d 3
```

We can see the partitions number was recognized correctly.

![image](https://user-images.githubusercontent.com/18204803/150090107-5b82b6a3-c5c6-4e9a-9163-53b2aa273cc3.png)

Reply:
I also tried `-d '{"partitions":"3","properties":{"k1":"v1","k2":"v2"}}'` and it works well.

Reply:
I have tried `curl -L http://localhost:8080/admin/v2/persistent/public/default/xxx/partitions -X PUT -H 'Content-Type: application/json' -d '{"partitions":"3","properties":{"k1":"v1","k2":"v2"}}'`

Got 400 at the broker side:

```
2022-01-19T18:41:56,584+0800 [pulsar-web-59-12] INFO  org.eclipse.jetty.server.RequestLog - 127.0.0.1 - - [19/Jan/2022:18:41:56 +0800] "PUT /admin/v2/persistent/public/default/xxx/partitions HTTP/1.1" 400 212 "-" "curl/7.77.0" 2
```

Got exception at the client-side

```
Cannot deserialize value of type `int` from Object value (token `JsonToken.START_OBJECT`)
 at [Source: (org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream); line: 1, column: 1]%
```

Reply:
If we can't avoid breaking compatibility, we can add a new endpoint in `v3` REST APIs

Reply:
Did you compile the latest code?

```bash
$ curl -L http://localhost:8080/admin/v2/persistent/public/default/xxx/partitions -X PUT -H 'Content-Type: application/json' -d '{"partitions":"3","properties":{"k1":"v1","k2":"v2"}}'
$ curl -L http://localhost:8080/admin/v2/persistent/public/default/xxx/partitions                                                                                                      
{"partitions":3,"properties":{"k1":"v1","k2":"v2"}}%                                                                                                                                                                                                          $ curl -L http://localhost:8080/admin/v2/persistent/public/default/xxx/partitions -X DELETE
$ curl -L http://localhost:8080/admin/v2/persistent/public/default/xxx/partitions -X PUT -H 'Content-Type: application/json' -d '{"partitions":"3","properties":{"k3":"v3"}}'          
$ curl -L http://localhost:8080/admin/v2/persistent/public/default/xxx/partitions                                                                                            
{"partitions":3,"properties":{"k3":"v3"}}
```

If you used the original code, since the body must be an integer, the body `{"partitions":"3","properties":{"k1":"v1","k2":"v2"}}` cannot be recognized.

```bash
$ curl -L http://localhost:8080/admin/v2/persistent/public/default/xxx/partitions -X PUT -H 'Content-Type: application/json' -d '3'                 
$ curl -L http://localhost:8080/admin/v2/persistent/public/default/xxx/partitions                                                                                                      
{"partitions":3}%                                                                                                                                                                                                                                             $ curl -L http://localhost:8080/admin/v2/persistent/public/default/xxx/partitions -X PUT -H 'Content-Type: application/json' -d '{"partitions":"3"}'
Cannot deserialize value of type `int` from Object value (token `JsonToken.START_OBJECT`)
 at [Source: (org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream); line: 1, column: 1]
```

But I don't think it breaks the backward compatibility. This PR makes the REST API available for more input.

Reply:
> If you used the original code, since the body must be an integer, the body {"partitions":"3","properties":{"k1":"v1","k2":"v2"}} cannot be recognized.

Yes, this is my test case, use the new Pulsar Admin version to point to the old broker.

Reply:
Got it. I think there is no good way for this compatibility. We can only guarantee **New Broker is compatible with Old Client**